pytrip/vdx.py

```diff
@@ -531,21 +551,35 @@ def create_sphere(cube, name, center, radius):
     :returns: A new Voi object.
     """
     v = Voi(name, cube)
-    t = np.linspace(start=0, stop=2.0 * pi, num=100)
+
+    t = np.linspace(start=0, stop=2.0 * pi,
+                    num=99, endpoint=False)  # num: sets the number of corners in sphere per slice.
     p = list(zip(np.cos(t), np.sin(t)))
+
+    points = []
+
     for i in range(0, cube.dimz):
         z = i * cube.slice_distance
         if center[2] - radius <= z <= center[2] + radius:
-            r = (radius**2 - (z - center[2])**2)**0.5
+            r2 = radius**2 - (z - center[2])**2
             s = Slice(cube)
             s.thickness = cube.slice_distance
-            if r > 0:
-                points = [[center[0] + r * x[0], center[1] + r * x[1], z] for x in p]
-            else:
-                points = [[center[0], center[1], z]]
+            if r2 > 0.0:
```

Comment:
Something is badly broken around this place. Tests for all version fail, see i.e. https://travis-ci.org/pytrip/pytrip/jobs/296438337

```
 def test_create_voi_sphere(self):
        logger.info("Creating CT cube from path " + self.cube000)
        c = pt.CtxCube()
        c.read(self.cube000)
        logger.info("Generating and adding sphere VOI")
        v = create_sphere(c, name="cube3", center=[10, 10, 10], radius=8)
        (...)
        
        contour = s.contour[0]
    
        logger.info("Checking Contour calculate_center method")
        center, area = contour.calculate_center()
>       self.assertAlmostEqual(center[0], 10.0)
E       AssertionError: 9.9211186575976953 != 10.0 within 7 places
```